pkg/ingestor/puller/blob/blob.go

```diff
@@ -97,6 +98,41 @@ func (bs *BlobStore) openBucket(ctx context.Context) (*blob.Bucket, error) {
 	return bucket, nil
 }
 
+func (bs *BlobStore) listFiles(ctx context.Context, b *blob.Bucket, prefix string, recursive bool) ([]*puller.ListObject, error) {
```

Comment:
It would be nice to have a test function for recursive functions though (and to keep code coverage high enough not to be ashamed of it 👀)

Reply:
added a bunch of unit tests